jupyterhub/proxy.py

```diff
@@ -492,12 +492,18 @@ def _api_url_default(self):
     )
 
     def _check_pid(self, pid):
-        if os.name == 'nt':
-            import psutil
+        import psutil
```

Comment:
psutil is only a dependency on Windows, which is why it's only imported on the Windows paths. It's a nontrivial compiled dependency with no wheels on non-Windows, so I'd prefer to avoid it as a hard dependency if we can.

Reply:
Unfortunately I'm not seeing an API in plain `os` which can get process cmdline. One option is to make the check only happen on the Windows path. This will only fix this issue on Windows and not Linux... this is still an improvement since this issue is more prevalent on Windows anyway.

Another option for avoiding psutil on Linux may be to use `subprocess` to call into utilities like `ps`. However, this may also not be a great choice, since we aren't guaranteed that they will be installed. This also introduces significant divergence into the windows/linux code paths.

According to https://github.com/giampaolo/psutil/issues/824, https://github.com/giampaolo/psutil/issues/1550, https://github.com/giampaolo/psutil/issues/1864 there may actually be wheels for psutil soon. So perhaps the best solution for now is the first one - keep the psutil usage in the Windows path, and we can propagate the fix to the Linux code path once wheels are provided?

Reply:
Yes, I think that's a good idea. Windows is not officially supported, but we are accepting contributions to fix issues, so keeping this to the Windows-only path seems sensible to me.